File: storagemanager/src/main/java/com/example/alexeyglushkov/cachemanager/StorageCleaner.java
package com.example.alexeyglushkov.cachemanager;

import com.example.alexeyglushkov.streamlib.progress.ProgressUpdater;

import io.reactivex.Completable;

/**
 * Created by alexeyglushkov on 10.10.15.
 */
public interface StorageCleaner {
    void setProgressInfo(ProgressUpdater info);
    Completable clean(Storage provider);
}

File: storagemanager/src/main/java/com/example/alexeyglushkov/cachemanager/StorageCleaner.kt
package com.example.alexeyglushkov.cachemanager

import com.example.alexeyglushkov.streamlib.progress.ProgressUpdater
import io.reactivex.Completable

/**
 * Created by alexeyglushkov on 10.10.15.
 */
interface StorageCleaner {
    fun setProgressInfo(info: ProgressUpdater?)
    fun clean(storage: Storage): Completable
    fun clean(storage: RxStorage): Completable
}